### DISCUSSION TITLE
Inline #stack(), and need #overarc()

### DISCUSSION THREAD

**Discussion by onomou** [typst/typst#3563] [Ideas]
My end goal is to reproduce `$\overarc{AC}$` from LaTeX. In the process, I tried to use #stack() like this:

```
Then, starting at A, he drew #stack("⌢","DE") with center A, #stack("⌢","EF") with center B...
```
![image](https://github.com/typst/typst/assets/131693/762beadd-c68e-4b16-a4d7-9bdc03f28951)

It should look like this:

![image](https://github.com/typst/typst/assets/131693/86a44c91-05c4-4b5b-bf46-9300ba6fdb08)

This is two separate issues. First, Typst needs a proper `#overarc()`. Second, `#stack()` needs an inline option. Existing [over/under commands](https://typst.app/docs/reference/math/underover) -- `underline`, `overline`, `underbrace`, `overbrace`, `underbracket`, and `overbracket` -- are insufficient.

**Comment by frozolotl** [comment]
You can use `box` for making things inline:
```typ
#let overarc(body) = box(stack("⌢", body))
Then, starting at A, he drew #overarc[DE] with center A, #overarc[EF] with center B...
```
![the above code, rendered](https://github.com/typst/typst/assets/44589151/89f544b6-4e68-4745-a0eb-4e9768fbe732)

**Reply by onomou** [reply]
Good idea with `box`. I have a lot to learn about finding information in the docs.

> The box function can be used to integrate such elements into a paragraph.
https://typst.app/docs/reference/layout/box/

**Comment by PgBiel** [comment]
Regarding `#overarc()`, usually `accent`[^1] and related functions are used for this; perhaps you can suggest a new one in an issue? :)

Edit: check out https://github.com/typst/typst/issues/2404

Edit: this seems to work (from a comment there):

```typ
$ accent(A B, paren.t) $
```

or
```typ
#let overarc(body) = math.accent(body, sym.paren.t)

$ overarc(A B) $
```

[^1]: https://typst.app/docs/reference/math/accent/

**Reply by onomou** [reply]
That works for me too. Thank you for the suggestion, and for showing me `math.accent`. I hadn't found that in the documentation yet. Also looks like I missed #2404 in my searches. But honestly, this is starting to feel like LaTeX all over again. What are the chances of baking in `#overarc# like they have `#overbrace` etc.? Or merging `accent` with `under/over`?

**Reply by PgBiel** [reply]
Not sure what you mean by "feel like LaTeX", it's just that there isn't a built-in function for your particular case (and that's going to happen pretty often - the standard library can't contain everything).
But even then, as you saw, it is pretty easy to create your own function building upon what the standard library has. Note that `sym` is only a small collection of built-in symbols, but Typst works with any Unicode symbols! (Of course, the `accent` function doesn't work with any symbol, but it uses Unicode symbols.) So you can also just take any symbols missing in Typst and paste them directly into your document for example (you could have placed `sym.paren.t` literally there as a string).

So I'm not sure the comparison is very apt.

> What are the chances of baking in #overarc# like they have #overbraceetc.?

If there is enough demand for it, it could be considered.

> Or merging accent with under/over`?

That sounds like a more significant proposal. I suggest laying out your ideas in the Math forge thread in the Discord server.

**Reply by frozolotl** [reply]
I did create a pull request yesterday that at least adds the `paren.t("AB")` syntax: https://github.com/typst/typst/pull/3564
I might do one for a nicer alias as well.

**Comment by emilyyyylime** [comment]
Another option is to use the unicode combining overarch character `͡`

**Reply by frozolotl** [reply]
That is a double-inverted breve, and it typically doesn't scale with the size. If that is something you want, then it can be a reasonable choice, though.
